packages/fluentui/react-northstar/src/components/Carousel/CarouselPaddle.tsx

```diff
@@ -0,0 +1,201 @@
+  disabled?: boolean;
+
+  /**
+   * Called after a user clicks the button.
+   * @param event - React's original SyntheticEvent.
+   * @param data - All props.
+   */
+  onClick?: ComponentEventHandler<CarouselPaddleProps>;
+
+  /**
+   * Called after a user focuses the button.
+   * @param event - React's original SyntheticEvent.
+   * @param data - All props.
+   */
+  onFocus?: ComponentEventHandler<CarouselPaddleProps>;
+
+  /** A paddle can indicate that it slides to the next item. */
+  next?: boolean;
+
+  /** A paddle can indicate that it slides to the previous item. */
+  previous?: boolean;
+}
+
+export type CarouselPaddleSlotClassNames = {
+  content: string;
+};
+
+export type CarouselPaddleStylesProps = Pick<CarouselPaddleProps, 'disabled' | 'next' | 'previous'>;
+
+const CarouselPaddle: React.FC<WithAsProp<CarouselPaddleProps>> &
+  FluentComponentStaticProps<CarouselPaddleProps> & { slotClassNames: CarouselPaddleSlotClassNames } = props => {
+  const context: ProviderContextPrepared = React.useContext(ThemeContext);
+  const { setStart, setEnd } = useTelemetry(CarouselPaddle.displayName, context.telemetry);
+  setStart();
+
+  const {
+    accessibility,
+    // @ts-ignore
+    active,
+    as,
+    children,
+    content,
+    disabled,
+    className,
+    next,
+    previous,
+    styles,
+    variables,
+    design,
+  } = props;
+
+  const hasChildren = childrenExist(children);
+
+  const getA11Props = useAccessibility(accessibility, {
+    debugName: CarouselPaddle.displayName,
+    mapPropsToBehavior: () => ({
+      as,
+      active,
+      disabled,
+    }),
+    actionHandlers: {
+      performClick: event => {
+        event.preventDefault();
+        handleClick(event);
+      },
+    },
+    rtl: context.rtl,
+  });
+  const { classes, styles: resolvedStyles } = useStyles<CarouselPaddleStylesProps>(CarouselPaddle.displayName, {
+    className: CarouselPaddle.className,
+    mapPropsToStyles: () => ({
+      disabled,
+      next,
+      previous,
+    }),
+    mapPropsToInlineStyles: () => ({
+      className,
+      design,
+      styles,
+      variables,
+    }),
+    rtl: context.rtl,
+  });
+
+  const unhandledProps = useUnhandledProps(CarouselPaddle.handledProps, props);
+  const ElementType = getElementType(props);
+
+  const handleClick = (e: React.SyntheticEvent) => {
+    if (disabled) {
+      e.preventDefault();
+      return;
+    }
+
+    _.invoke(props, 'onClick', e, props);
+  };
+
+  const handleFocus = (e: React.SyntheticEvent) => {
+    _.invoke(props, 'onFocus', e, props);
+  };
+
+  const result = (
+    <ElementType
+      {...rtlTextContainer.getAttributes({ forElements: [children] })}
+      {...getA11Props('root', {
+        onClick: handleClick,
+        disabled,
+        className: classes.root,
+        onFocus: handleFocus,
+        ...unhandledProps,
+      })}
+    >
+      {hasChildren
+        ? children
+        : Box.create(content, {
```

Comment:
Do we need this? 😭

Reply:
The reason is a bit stupid. I need to use the ':before' element eitherwise for setting the chevron, but guess what, it is used for the focusing styles :( :( :(